crates/ln-dlc-node/src/node/invoice.rs

```diff
@@ -103,6 +103,19 @@ where
             })
             .map_err(|_| anyhow!("Failed to sign invoice"))?;
         let invoice = Invoice::from_signed(signed_invoice)?;
+
+        self.storage.insert_payment(
+            payment_hash,
+            PaymentInfo {
+                preimage: Some(
+                    self.channel_manager
+                        .get_payment_preimage(payment_hash, payment_secret)
+                        .map_err(|_| anyhow!("Failed to get payment preimage for new invoice"))?,
```

Comment:
🤔 I'm not sure it's such a good idea to store the preimage from the perspective of the invoice generator. If this is leaked before the final HTLC is claimed then people can lose money. I would leave this as `None` for now and consider removing it from the model altogether.

Reply:
Fair enough - I just added it because it was on the Figma mock - @bonomat thoughts?

Reply:
I'll change it to use the preimage from PaymentClaimed event - is this sufficient? If I understand correctly, this is after funds have been transferred, so it should be fine (?)